File: src/main/java/com/ddd/trainings/idea/api/http/TrainingIdeaRestController.java
package com.ddd.trainings.idea.api.http;

import com.ddd.trainings.idea.api.app.TrainingIdeaService;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class TrainingIdeaRestController {

  private final TrainingIdeaService service;

  public TrainingIdeaRestController(TrainingIdeaService service) {
    this.service = service;
  }

  @PostMapping("/training/ideas")
  public void createTrainingIdea(String trainerId, String name) {
    service.createTrainingIdea(trainerId, name);
  }

  @PostMapping("/training/ideas/{ideaId}/propose")
  public void proposeTrainingIdea(String ideaId) {
    service.proposeIdea(ideaId);
  }
}

File: src/Ddd/Trainings/Idea/Adapters/TrainingIdeaRestController.cs
using Ddd.Trainings.Idea.Api.App;
using Microsoft.AspNetCore.Mvc;

namespace Ddd.Trainings.Idea.Api.Http
{
    [ApiController]
    [Route("training/ideas")]
    public class TrainingIdeaRestController : ControllerBase
    {
        private readonly TrainingIdeaService _service;

        public TrainingIdeaRestController(TrainingIdeaService service)
        {
            _service = service;
        }

        [HttpPost]
        public IActionResult CreateTrainingIdea(string trainerId, string name)
        {
            _service.CreateTrainingIdea(trainerId, name);
            return Ok();
        }

        [HttpPost("{ideaId}/propose")]
        public IActionResult ProposeTrainingIdea(string ideaId)
        {
            _service.ProposeIdea(ideaId);
            return Ok();
        }
    }
} 